src/test/java/spoon/MavenLauncherTest.java

```diff
@@ -0,0 +1,35 @@
+package spoon;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class MavenLauncherTest {
+	@Test
+	public void spoonMavenLauncherTest() {
+		// without the tests
+		MavenLauncher launcher = new MavenLauncher("./", MavenLauncher.SOURCE_TYPE.APP_SOURCE);
+
+		assertEquals(8, launcher.getEnvironment().getComplianceLevel());
+
+		assertEquals(10, launcher.getEnvironment().getSourceClasspath().length);
+		// 54 because of the sub folders of src/main/java
+		assertEquals(54, launcher.getModelBuilder().getInputSources().size());
+
+		// with the tests
+		launcher = new MavenLauncher("./", MavenLauncher.SOURCE_TYPE.ALL_SOURCE);
+		assertEquals(10, launcher.getEnvironment().getSourceClasspath().length);
+		// 235 because of the sub folders of src/main/java and src/test/java
+		assertEquals(235, launcher.getModelBuilder().getInputSources().size());
+	}
+
+	@Test(expected = SpoonException.class)
+	public void MavenLauncherOnFileTest() {
+		new MavenLauncher("./pom.xml", MavenLauncher.SOURCE_TYPE.APP_SOURCE);
```

Comment:
Why this behaviour? Why only accepting the root directory and not the pom.xml itself?

Reply:
because it's boring to handle a lot of different cases.

Reply:
It seems pretty easy (and useful IMHO) to handle this one...